### PR TITLE
pace-util release v0.7.0

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

This PR adds release instructions for pypi to pace-util, and prepares it for v0.7.0 release.

## Infrastructure changes:

- Added RELEASE.rst instructions and `make release` target in pace-util

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated

### CODE REVIEW

File: pace-util/Makefile [hunk 1]
```diff
@@ -51,28 +51,15 @@ lint:
 	mypy --follow-imports silent --show-error-codes fv3gfs
 	@echo "LINTING SUCCESSFUL"
 
-# We do not use --no-annotate to keep the dependencies annotations
```
Comment (mcgibbon) [inline pace-util/Makefile:54]: pace-util uses the top level constraints file via a symbolic link, this target would overwrite that symlink and should be removed.

File: pace-util/RELEASE.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+3. Update the HISTORY.md, replacing the "latest" version heading with the new version.
+
+4. Commit your changes so far to the release branch.
+
+5. In the pace-util directory, run `bumpversion <major/minor/patch>`. This will create a new commit.
+
+6. `git push -u origin release/util/<version>` and create a new pull request in Github.
+
+7. When the pull request is merged to master, `git checkout master` and `git pull`,
```
Comment (mcgibbon) [inline pace-util/RELEASE.rst:19]:
```suggestion
7. When the pull request is merged to main, `git checkout main` and `git pull`,
```

**Comment by rheacangeo** [comment]
launch jenkins

File: pace-util/Makefile (hunk 1 above)
Comment (mcgibbon) [reply pace-util/Makefile:54]: I was wrong - this must be in a branch. Reverting this removal.

File: pace-util/HISTORY.md [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
+- Added HaloUpdater and associated code, which compiles halo packing for more efficient halo updates
+- Added physical constants to pace.util.constants
+
+Minor changes:
+- Added method set_extra_dim_lengths to QuantityFactory
 
 Fixes:
 - Fixed bug where ZarrMonitor depended on dict `.items()` always returning items in the same order
 
+Other changes may exist in this version, as we were not careful about updating the history on each PR.
```
Comment (mcgibbon) [inline pace-util/HISTORY.md:20]:
```suggestion
Other changes may exist in this version, as we temporarily paused updating the history on each PR.
```

**Review by spencerkclark — APPROVED** [review]
Looks good to me

**Review by rheacangeo — APPROVED** [review]
Awesome! Thanks for providing such great instructions. Looks like you have a wording change suggestion for yourself.

**Pull request merged**